### DISCUSSION TITLE
Channel[x] reading ESP32 Script

### DISCUSSION THREAD

**Discussion by mmmmm009** [arendst/Tasmota#20873] [Q&A] [answered]
Hi,
I have ESP32 with Scripting (tasmo compiler Script option selected, no additional #define)

```
>D
chan1=0
>E
chan1=Channel[2]
print %chan1%
```

Why do I keep getting: ?

```
15:09:01.939 var not found: Channel[2]
```
(Channel 2 is physically available)
I've tried >E >S >F

**Comment by sfromis** [comment, marked as answer]
Assuming that you want the PWM `Channel2` value, those are not indexed in an array, therefore you should not have those `[]`

**Comment by HofeBY** [comment]
@mmmmm009

can you show us the Tasmota logging & configuration of GPIO settings?

**Reply by mmmmm009** [reply]
`Channel2` instead of `Channel[2]` did the trick
however it's contradictory to:
```
; or check change of color channels
chan1=Channel[1]
chan2=Channel[2]
chan3=Channel[3]
```
(above example from: [https://tasmota.github.io/docs/Scripting-Language/#scripting-language-example](url))

**Reply by sfromis** [reply]
That's why I explicitly stated my assumption about it being the `Channel2` value from a PWM-oriented payload, which appears to be a correct guess. Here, the format in the JSON is like `"Channel3":57`

The example you quote is **not** for such a payload, but instead for color channels of a payload for a lamp/bulb, where they appear in a format like `"Channel":[13,20,27,0,0]` indeed an array.

These two things are not the same, even if both have a field name with "Channel".

In general, you have to read the JSON payload you want values from, and strictly follow the structure of that.
